src/crypto/openssl/cose_signer.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "crypto/openssl/cose_signer.h"
+
+#include "ccf/ds/logger.h"
+
+#include <openssl/evp.h>
+#include <t_cose/t_cose_sign1_sign.h>
+
+namespace ccf::crypto
+{
+  using namespace OpenSSL;
+
+  COSESigner_OpenSSL::COSESigner_OpenSSL(const Pem& key) :
+    key_pair(std::make_shared<KeyPair_OpenSSL>(key))
```

Comment:
Unless I am missing something, we have a shared ptr to the key where we intend do this, we do not want to re-parse and make a copy (not so much for performance, we presumably create the signer once, but to avoid having the key more than once).

Reply:
Used EVP*